datatree/tests/test_datatree.py

```diff
@@ -693,3 +693,42 @@ def test_filter(self):
         )
         elders = simpsons.filter(lambda node: node["age"] > 18)
         dtt.assert_identical(elders, expected)
+
+
+class TestSubtreeMerge:
+    def test_merge_subtrees(self):
+        data_tree = xr.DataArray(
+            np.random.rand(3, 3, 3),
+            coords=[("x", [0, 1, 2]), ("y", [0, 1, 2]), ("z", [0, 1, 2])],
+            dims=["x", "y", "z"],
+        ).groupby("x")
+        merged_data_tree = data_tree.merge_subtrees(subtree_a=0, subtree_b=1)
+
+        self.assertEqual(merged_data_tree.dims, ("x", "y", "z"))
+        self.assertEqual(list(merged_data_tree.coords.keys()), ["x", "y", "z"])
+        self.assertListEqual(list(merged_data_tree.coords["x"].values), [0, 2])
+
+        expected_values = np.concatenate(
+            [data_tree.get_group(0), data_tree.get_group(1)], axis=0
```

Comment:
I'm confused by this - even if your `data_tree` variable were actually an instance of the `DataTree` class, there still is not actually a method called `get_group` defined on `DataTree`. You can see this by trying to call it in ipython:

```python
In [1]: from datatree import DataTree

In [2]: dt = DataTree()

In [3]: dt.get_group
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
Cell In[3], line 1
----> 1 dt.get_group

File ~/Documents/Work/Code/datatree/datatree/common.py:54, in TreeAttrAccessMixin.__getattr__(self, name)
     52         with suppress(KeyError):
     53             return source[name]
---> 54 raise AttributeError(
     55     f"{type(self).__name__!r} object has no attribute {name!r}"
     56 )

AttributeError: 'DataTree' object has no attribute 'get_group'
```

The actual object you have created here is a [`DataArrayGroupby` object](https://docs.xarray.dev/en/stable/generated/xarray.core.groupby.DataArrayGroupBy.html#xarray.core.groupby.DataArrayGroupBy), but this doesn't have a `get_group` method either.